### PR TITLE
Make sure tokenHandler exists, don't call API when still fetching

### PR DESCRIPTION
**Pull request by putrikarunia**
- Make sure tokenHandler is passed into each class.
- Only call api to refresh token once, all other requests to refresh the token will wait for this one api call.

### CODE REVIEW

File: src/handler/TokenHandler.ts [hunk 1]
```diff
@@ -80,18 +91,50 @@ class TokenHandler {
...
+        ) {
+          this.tokenFetchingState = TOKEN_FETCHING_STATES.initial;
+          reject();
+          return;
+        } else if (
+          this.tokenFetchingState === TOKEN_FETCHING_STATES.errorFatal
+        ) {
+          reject();
+          return;
+        }
```
Comment (albertpurnama) [inline src/handler/TokenHandler.ts]: can you put this conditional logic in a switch statement for better readability?

File: src/Cotter.ts [hunk 2]
```diff
@@ -76,21 +76,27 @@ export default class Cotter extends CotterVerify {
...
+        this.tokenHander
+      );
+      return await web.show();
+    }
+    let web = new WebAuthn(
+      {
         ApiKeyID: this.config.ApiKeyID,
         Identifier: identifier,
         Type: "REGISTRATION",
-        ErrorDisplay: "The browser or user device doesn't support WebAuthn.",
```
Comment (albertpurnama) [inline src/Cotter.ts:83]: why is the error message gets deleted here?

Comment (albertpurnama) [reply src/Cotter.ts:83]: wait nvm, it was not there in the first place.

File: src/handler/TokenHandler.ts [hunk 3]
```diff
@@ -80,18 +91,50 @@ class TokenHandler {
...
+          this.tokenFetchingState === TOKEN_FETCHING_STATES.ready &&
+          this.fetchTokenResp
+        ) {
+          resolve(this.fetchTokenResp);
+          return;
+        } else if (
+          this.tokenFetchingState === TOKEN_FETCHING_STATES.errorRetry
+        ) {
+          this.tokenFetchingState = TOKEN_FETCHING_STATES.initial;
+          reject();
```
Comment (albertpurnama) [inline src/handler/TokenHandler.ts]: if it's errorRetry shouldn't it retry to fetch the token? why reject and return?

Comment (putrikarunia) [reply src/handler/TokenHandler.ts]: errorRetry means if `getAccessToken` or `getIdToken` gets called again, can retry to refetch token. But we don't want to automatically refetch here cos it keeps on erroring with the same error then it'll keep on retrying forever

Comment (albertpurnama) [reply src/handler/TokenHandler.ts]: I see, so the logic for retries should be handled by the user of the SDK?

Comment (putrikarunia) [reply src/handler/TokenHandler.ts]: Yep, most probably would involve the end-user pressing smth

**Review by albertpurnama — APPROVED** [review]

**Pull request merged**